blog/2016-04-20-futhark-and-pygame.rst

```diff
@@ -0,0 +1,257 @@
+---
+title: Creating Interactive Futhark GUIs
+author: Niels G. W. Serup
+description: Using PyGame and PyOpenCL to interact with Futhark programs in simple graphical user interfaces.
+---
+
+In `the previous blog post
+</blog/2016-04-15-futhark-and-pyopencl.html>`_, Troels showed you how
+to use Futhark's PyOpenCL backend to generate PNG files with Python
+and NumPy.  There was also a short mention of *interactive* examples
+built with PyGame.  In this blog post we will demonstrate how to add
+interactive visual interfaces to simulations implemented in Futhark.
+
+`PyGame <http://www.pygame.org/>`_ is a Python library for the SDL 1.2
+graphics framework.  Although a Python 3 port does exist, for
+compatibility reasons we use the original Python 2 library.
+Library-savvy readers might point out that a more modern library like
+pyglet or PySDL2 might be more... modern, but we really only require
+few features, so good old PyGame is sufficient for our basic examples.
+
+We currently have four GUI examples:
+
+  + `Mandelbrot Explorer <https://github.com/HIPERFIT/futhark-benchmarks/tree/master/misc/mandelbrot-explorer>`_
+  + `Fluid Simulation GUI <https://github.com/HIPERFIT/futhark-benchmarks/tree/master/accelerate/fluid/gui>`_
+  + `Game of Life <https://github.com/HIPERFIT/futhark-benchmarks/tree/master/misc/life>`_
+  + `HotSpot 2D Heat Equation GUI <https://github.com/HIPERFIT/futhark-benchmarks/tree/master/rodinia/hotspot>`_
+
+It is pretty easy to create an interactive visualisation with Futhark,
+so expect this number to go up.  However, note that Futhark
+development does *not* target interactive visualisations -- we just
```

Comment:
does *not* target interactive visualisations => is not primarily oriented towards graphics programming